core/sql/optimizer/SearchKey.cpp

```diff
@@ -2711,6 +2716,65 @@ void SearchKey::computeCoveredLeadingKeys()
   }
 }
 
+void SearchKey::computeFullKeyPredicates(ValueIdSet& predicates)
+{
+  // If the key column is an INDEX COLUMN, convert it into a base column
+  ValueIdList baseKeyCols;
+
+  ValueId vid;
+  const ValueIdList& keyColumns = getKeyColumns();
+  for (CollIndex i=0; i<keyColumns.entries(); i++ ) {
+     if ( keyColumns[i].getItemExpr()->getOperatorType() == ITM_INDEXCOLUMN )
+       vid = ((IndexColumn*)(keyColumns[i].getItemExpr()))->getDefinition();
+     else
+       vid = keyColumns[i];
+
+     baseKeyCols.insertAt(baseKeyCols.entries(), vid);
+  }
+
+  // compute the # of leading key columns covered by the key columns 
+  // in the disjunct.
```

Comment:
Not sure I understand this comment since I don't see a counter being incremented anywhere. I do see the fullKeyPredicates_ member being populated. I'm guessing the intent is to get the covered leading prefix of the key. The correctness of the code I'm guessing depends on "predicates" being ordered by key position but "predicates" is a ValueIdSet, so does it have an order?